core/services/validators.go

```diff
@@ -378,42 +376,63 @@ func ValidateServiceAgreement(sa models.ServiceAgreement, store *store.Store) er
 }
 
 // ValidatedOracleSpecToml validates an oracle spec that came from TOML
-func ValidatedOracleSpecToml(tomlString string) (offchainreporting.OracleSpec, error) {
-	var m toml.MetaData
-
-	// Sane defaults
-	spec := offchainreporting.OracleSpec{
-		OffchainReportingOracleSpec: models.OffchainReportingOracleSpec{
-			P2PBootstrapPeers:                      pq.StringArray{},
-			ObservationTimeout:                     models.Interval(10 * time.Second),
-			BlockchainTimeout:                      models.Interval(20 * time.Second),
-			ContractConfigTrackerSubscribeInterval: models.Interval(2 * time.Minute),
-			ContractConfigTrackerPollInterval:      models.Interval(1 * time.Minute),
-			ContractConfigConfirmations:            uint16(3), // TODO: why a uint16? just forcing casting everywhere
-		},
+func ValidatedOracleSpecToml(tomlString string) (spec offchainreporting.OracleSpec, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = errors.Errorf("panicked with err %v", r)
+		}
+	}()
+
+	var oros models.OffchainReportingOracleSpec
+	spec = offchainreporting.OracleSpec{
 		Pipeline: *pipeline.NewTaskDAG(),
 	}
-	m, err := toml.Decode(tomlString, &spec)
+	tree, err := toml.Load(tomlString)
+	if err != nil {
+		return spec, err
+	}
+	err = tree.Unmarshal(&oros)
 	if err != nil {
 		return spec, err
 	}
+	err = tree.Unmarshal(&spec)
+	if err != nil {
+		return spec, err
+	}
+	spec.OffchainReportingOracleSpec = oros
+
+	// TODO: upstream a way to check for undecoded keys in go-toml
+	// TODO: upstream support for time.Duration defaults in go-toml
```

Comment:
Mind making stories for these in Tracker? Hard to prioritize against other work when they're just mixed in the codebase, and they will often get forgotten.

[Typically](https://github.com/smartcontractkit/chainlink/blob/develop/tools/ci-ts/test-helpers/chainlinkClient.ts#L3) we drop a Tracker link below a short description.